packages/perseus-editor/src/components/widget-editor.tsx

```diff
@@ -72,14 +76,15 @@ class WidgetEditor extends React.Component<
         cb: () => unknown,
         silent: boolean,
     ) => {
+        if (!this.widget.current) {
+            return;
+        }
         const newWidgetInfo = Object.assign(
             {},
             this.state.widgetInfo,
         ) as PerseusWidget;
         newWidgetInfo.options = Object.assign(
-            // eslint-disable-next-line react/no-string-refs
-            // @ts-expect-error - TS2339 - Property 'serialize' does not exist on type 'ReactInstance'.
-            this.refs.widget.serialize(),
+            this.widget.current.serialize(),
```

Comment:
Nice!  I like that we're switching to an actual ref. 

If we avoid calling `props.onChange` when the widget ref isn't set up.  Will that mean we don't propagate that initial TeX mixup up to the parent editors? Or is that the part that you said didnt' work quite as I thought? 

I'll approve and let you make the final call based on what worked for you.

Reply:
We couldn't have done, because that onChange function relies on having the serialized widget. If I swap out for an empty object or something we'd REALLY muck up the data.

Yeah, we don't propagate the initial TeX mixup... which is what I thought you wanted